plonky2x/src/frontend/builder/mux.rs

```diff
@@ -9,7 +9,22 @@ trait MuxBuilder {
 
 impl<F: RichField + Extendable<D>, const D: usize> MuxBuilder for CircuitBuilder<F, D> {
     fn select_index<T: CircuitVariable>(&mut self, selector: Variable, inputs: &[T]) -> T {
-        todo!()
+        let num_var_in_t = inputs[0].variables().len();
+        let input_len = inputs.len();
+        let mut res = (0..num_var_in_t).map(|_| self.init::<Variable>()).collect::<Vec<_>>();
```

Comment:
Here you're basically just allocating the variables for the result. I think it's better to not do this and instead just allocate an empty `Vec` and `push` to the vec, which will be easier after you use the `random_access` gadget that I wrote the comment about below